extensions/cornerstone/src/initContextMenu.ts

```diff
@@ -0,0 +1,132 @@
+import { eventTarget, Types, EVENTS } from '@cornerstonejs/core';
+import { Enums } from '@cornerstonejs/tools';
+import { setEnabledElement } from './state';
+
+const cs3DToolsEvents = Enums.Events;
+
+const showContextMenuDefault: Types.CommandCustomization = {
+  commands: [
+    {
+      commandName: 'showViewerContextMenu',
+      commandOptions: {
+        menuName: 'cornerstoneContextMenu',
+      },
+      context: 'CORNERSTONE',
+    },
+  ],
+};
+
+function initContextMenu({
+  cornerstoneViewportService,
+  customizationService,
+  commandsManager,
+}): void {
+  /**
+   * Finds tool nearby event position triggered.
+   *
+   * @param {Object} commandsManager mannager of commands
+   * @param {Object} event that has being triggered
+   * @returns cs toolData or undefined if not found.
+   */
+  const findNearbyToolData = evt => {
+    if (!evt?.detail) {
+      return;
+    }
+    const { element, currentPoints } = evt.detail;
+    return commandsManager.runCommand(
+      'getNearbyToolData',
+      {
+        element,
+        canvasCoordinates: currentPoints?.canvas,
+      },
+      'CORNERSTONE'
+    );
+  };
+
+  const getShowContextMenu = (): Types.UICommand =>
+    customizationService.getModeCustomization(
+      'showContextMenu',
+      showContextMenuDefault
+    );
+
+  const showContextMenu = (contextMenuProps): void => {
+    customizationService.recordInteraction(
+      getShowContextMenu(),
+      contextMenuProps
+    );
+  };
```

Comment:
This requires explanation, I don't fully understand what you are doing here. So showContextMenu uses the customizationService to record interaction which uses the customization service to show context menu?

Reply:
Added explanation.